datafusion/core/tests/fuzz_cases/pruning.rs

```diff
@@ -0,0 +1,188 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use arrow_array::{ArrayRef, BooleanArray, StringArray};
+use arrow_schema::{DataType, Field, Schema};
+use datafusion::physical_optimizer::pruning::{PruningPredicate, PruningStatistics};
+use datafusion::prelude::*;
+use datafusion_common::ScalarValue;
+use datafusion_expr::{col, lit};
+use datafusion_physical_expr::planner::logical2physical;
+use rand::{thread_rng, Rng};
+use std::collections::HashSet;
+use std::sync::Arc;
+
+/// Tests for `LIKE` with truncated statistics to validate incrementing logic
+///
+/// Create several 2 row batches and ensure that `LIKE` with the min and max value
+/// are correctly pruned even when the "statistics" are trunated.
+#[test]
+fn test_prune_like_truncated_statistics() {
```

Comment:
I think it's also worth having tests for `=` and maybe other operators, it was not immediately obvious to me that there wasn't a bug with those as well.